server/src/main/java/org/elasticsearch/action/admin/cluster/repositories/verify/VerifyRepositoryRestResponse.java

```diff
@@ -0,0 +1,149 @@
+
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.Streamable;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class VerifyRepositoryRestResponse implements ToXContentObject {
+
+    public static class NodeView implements ToXContentObject {
+        private static final ObjectParser.NamedObjectParser<NodeView, Void> PARSER;
+        static {
+            ObjectParser<NodeView, Void> parser = new ObjectParser<>("nodes");
+            parser.declareString(NodeView::setName, new ParseField(Fields.NAME));
+            PARSER = (XContentParser p, Void v, String name )-> parser.parse(p, new NodeView(name), null);
+        }
+
+        final String nodeId;
+        String name;
+
+        public NodeView(String nodeId) { this.nodeId = nodeId; }
+
+        public NodeView(String nodeId, String name) {
+            this(nodeId);
+            this.name = name;
+        }
+
+        void setName(String name) { this.name = name; }
+
+        public String getName() { return name; }
+
+        public String getNodeId() { return nodeId; }
+
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject(nodeId);
+            builder.field(Fields.NAME, name);
+            builder.endObject();
+            return builder;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (obj == null) {
+                return false;
+            }
+            if (getClass() != obj.getClass()) {
+                return false;
+            }
+            NodeView other = (NodeView) obj;
+            return Objects.equals(nodeId, other.nodeId) &&
+                Objects.equals(name, other.name);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(nodeId, name);
+        }
+    }
+
+    private static final ObjectParser<VerifyRepositoryRestResponse, Void> PARSER =
+        new ObjectParser<>(VerifyRepositoryRestResponse.class.getName(), VerifyRepositoryRestResponse::new);
+    static {
+        PARSER.declareNamedObjects(VerifyRepositoryRestResponse::nodes, NodeView.PARSER, new ParseField("nodes"));
+    }
+
+    private List<NodeView> nodes = new ArrayList<>();
+
+    VerifyRepositoryRestResponse() {
+    }
+
+    public List<NodeView> nodes() {
+        return nodes;
+    }
+
+    void nodes(List<NodeView> nodes) {
+        this.nodes = nodes;
+    }
+
+    static final class Fields {
+        static final String NODES = "nodes";
```

Comment:
I believe we've fallen out of love with making these classes. Can you remove it? Personally I just use the string. Some folks make a string constant. Either way is fine though.